src/services/http-req/dogBreedRequest.test.ts

```diff
@@ -0,0 +1,71 @@
+import fetch from 'node-fetch'
+import { getDogBreeds } from './dogBreedRequest'
+import { input1 } from '../../lambdas/mock-events/mock-values'
+
+const mockedFetch: jest.Mock = fetch as any
+
+jest.mock('node-fetch')
+
+describe('fetching api data for dog breeds', () => {
+  const mockPayload = input1
+  beforeEach(() => {
+    mockedFetch.mockReturnValueOnce({
+      json: () => {
+        return { status: 'success', message: mockPayload }
+      },
+    })
+  })
+
+  it('should return the correct payload from a sucessful fetch request', async () => {
+    const response = await getDogBreeds()
+    expect(response.status).toBe('success')
+    expect(response.message).toMatchObject(input1)
+  })
+})
+
+describe('handling errors from the external call', () => {
+  beforeEach(() => {
+    mockedFetch.mockReturnValueOnce({
+      json: () => {
+        throw new Error('I broke somewhere')
+      },
+    })
+  })
+
+  it('should bubble up an error from the external API call', async () => {
+    let response
+    try {
+      response = await getDogBreeds()
+    } catch (err) {
+      response = err
+    }
+    expect(response instanceof Error).toBe(true)
+    expect(response.message).toEqual('I broke somewhere')
+  })
+})
```

Comment:
OK, that makes sense. I adjusted these tests to expect a rejection that equals an error. I do have a question about asserting errors. Is it better to run assertions on the error's name and message properties, or to run the assertion on the entire error?

Reply:
Typically we like to test the whole error, we try to keep the assertions to a minimum so something like this would be good:

`expect(promise).rejects.toThrowError(new Error('whoopsie'))`

It tests the message, the type, and everything in between, all in one assertion 👍 but no worries, what you have is good!

Reply:
Thank you for clarifying this for me it makes a lot of sense now